### ISSUE TITLE
[FEAT] Doc models factories

### ISSUE THREAD

**Issue by griest024** [graycoreio/daffodil#3326]
# :bulb: Feature request

## Feature Name

Doc models factories

## The Desired Behavior

Docsgen models should have associated factories

## Your Use Case

As a maintainer, I would like to have factories for docsgen models.

**Comment by griest024** [comment]
blocked by #3325 and #2819

**Comment by griest024** [comment]
@AmmarA06 this is ready for you. For now, just implement factories for the following models in `@daffodil/docs-utils`:
- DaffDocsApiDecorator
- DaffDocsApiRef
- DaffBreadcrumb
- DaffBaseDoc
- DaffPackageGuideDoc
- DaffDoc
- DaffDocsDesignExampleFile
- DaffDocsDesignExample
- DaffDocExample
- DaffApiNavDoc
- DaffApiNavPackageDoc
- DaffDesignGuideNavDoc
- DaffDocsNavList
- DaffDocsDesignGuideNavList
- DaffDocsApiNavList
- DaffApiPackageDoc
- DaffNavDoc
- DaffDocTableOfContentsEntry
Don't worry about the API role factories, that can be a separate PR.

When implementing the factories, you can refer to https://github.com/graycoreio/daffodil/blob/develop/libs/category/testing/src/factories/category.factory.ts as an example. In particular, pay attention to the fact that child factories are passed in through the constructor and the second type param of `DaffModelFactory` is set to the type of the mock model class: `typeof MockCategory`. This strongly types the super constructor call.

When types extend, you can extend the mock model classes like here: https://github.com/graycoreio/daffodil/blob/develop/libs/product-composite/testing/src/factories/composite-product.factory.ts.
